File: bench/Reference/file_readlines_large.js
const fs = require('fs');
const data = fs.readFileSync("bench/data/large_text.txt", "utf8");
let total = 0;
let start = 0;
for (let i = 0; i < data.length; i++) {
    if (data.charCodeAt(i) === 10) {
        total += (i - start);
        start = i + 1;
    }
}
if (start < data.length) total += data.length - start;
console.log(total);

File: bench/Reference/file_readlines_large.java
import java.io.*;

public class file_readlines_large {
    public static void main(String[] args) throws Exception {
        long total = 0;
        try (BufferedReader r = new BufferedReader(new FileReader("bench/data/large_text.txt"))) {
            String line;
            while ((line = r.readLine()) != null) total += line.length();
        }
        System.out.println(total);
    }
}
